diff --git a/ar2/lz1.c b/ar2/lz1.c
index 5aecc181..d4fde240 100644
--- a/ar2/lz1.c
+++ b/ar2/lz1.c
@@ -74,7 +74,7 @@ DCOMPTBL	*CrakTbl;
 
 int LZ_1(FILE *infile, FILE *outfile, long *bytes)
 	{
-	VOID				output();
+	VOID				output(int code, FILE *outfile);
 	WORD				c, ent, tag = TAG;
 	WORD				n_ent;
 	register COMPTBL	*ctp;
@@ -160,9 +160,7 @@ void addentry(WORD c, WORD ent)
  * Output the given code.
  */
 
-VOID	output(code, ofp)
-WORD	code;
-FILE	*ofp;
+VOID	output(int code, FILE *ofp)
 	{
 	if (code < 0)
 		{
@@ -262,15 +260,13 @@ void insert_bit(short code)
  * Decompress the input file.
  */
 
-WORD	de_LZ_1(infile, outfile, bytes)
-FILE	*infile, *outfile;
-long	bytes;
+WORD	de_LZ_1(FILE *infile, FILE *outfile, long bytes)
 	{
 	register DCOMPTBL	*dtp;
 	unsigned char			stack[MAXSTACK];
 	WORD				tag, finchar, code, oldcode, incode;
 	register unsigned char		*stackp = stack;
-	WORD				getcode();
+	WORD				getcode(FILE *infile);
 
 	if (readshort(infile, &tag) == EOF || tag != TAG)
 		return (NOT_AR);
@@ -323,8 +319,7 @@ long	bytes;
  * Read one code from the input file.  If EOF, return -1.
  */
 
-WORD	getcode(infile)
-FILE	*infile;
+WORD	getcode(FILE *infile)
 	{
 	WORD		code, reslt;
 	static WORD	size = 0;
