diff --git a/ar2/o2u.c b/ar2/o2u.c
index 9279a643..983d70b3 100644
--- a/ar2/o2u.c
+++ b/ar2/o2u.c
@@ -67,8 +67,7 @@ static int	monthLength[12] = {
         };
 
 
-char	*u2oDate(l)
-long	l;
+char	*u2oDate(time_t l)
 	{
 	static char	buf[5];
 	struct tm	*tp;
@@ -83,8 +82,7 @@ long	l;
 	}
 
 
-long	o2uDate(buf)
-char	*buf;
+long	o2uDate(char *buf)
 	{
 	register int	i;
 	int				year;
@@ -93,7 +91,6 @@ char	*buf;
 	int				hour;
 	int				minute;
 	long			result;
-	extern long		time();
 
 	int				timezone = 5*60*60;   /* I'm in the Eastern Time zone GMT + 5 */
 
@@ -181,8 +178,7 @@ char	*buf;
 	}
 
 
-char	u2oFmode(flags)
-short	flags;
+char	u2oFmode(mode_t flags)
 	{
 	char			os9Flag = 0;
 	register int	i;
@@ -201,8 +197,7 @@ short	flags;
 	}
 
 
-short	o2uFmode(flags)
-char	flags;
+short	o2uFmode(char flags)
 	{
 	short			unixFlag = 0;
 	register int	i;
