diff --git a/ar2/arsup.c b/ar2/arsup.c
index 43746307..f914569a 100644
--- a/ar2/arsup.c
+++ b/ar2/arsup.c
@@ -118,8 +118,7 @@ char	ch;
  *   back to a long.
  */
 
-long	c4tol(s)
-char	*s;
+long	c4tol(char *s)
 	{
 	long	x = 0;
 
@@ -195,11 +194,8 @@ void set_fstat(char *pn, FILDES *fs)
 	short			mode = o2uFmode(fs->fd_attr);
 	struct passwd	*pwdbuf;
 #if !defined(WIN32)
-	struct passwd	*getpwuid();
 #endif
-	struct  {
-		long	a, m;
-		} ubuf;
+	struct utimbuf ubuf;
 
 	s = (*p++&0xff);
 	s <<= 8;
@@ -212,8 +208,8 @@ void set_fstat(char *pn, FILDES *fs)
 	chown(pn, s, pwdbuf ? pwdbuf->pw_gid : s);
 #endif
 
-	ubuf.a = time((long *) 0);
-	ubuf.m = o2uDate(fs->fd_date);
+	ubuf.actime = time((long *) 0);
+	ubuf.modtime = o2uDate(fs->fd_date);
 	utime(pn, &ubuf);
 #else
 # ifdef OSK
@@ -232,8 +228,7 @@ void set_fstat(char *pn, FILDES *fs)
  * get the file size
  */
 
-long	get_fsize(pn)
-int		pn;
+long	get_fsize(int pn)
 	{
 #ifdef SYSV
 	struct stat	stbuf;
