diff --git a/mamou/mamou.h b/mamou/mamou.h
index 6a40b73e..da8d419a 100644
--- a/mamou/mamou.h
+++ b/mamou/mamou.h
@@ -18,6 +18,9 @@
 
 #include "cocopath.h"
 
+/* Forward declaration */
+typedef struct _assembler assembler;
+
 #define VERSION_MAJOR   1
 #define VERSION_MINOR   0
 
@@ -170,7 +173,7 @@ struct h6309_opcode
 	char		cycles;				/* base # of cycles (6809) */
 	char		plus_delta_6309;	/* +/- cycles to adjust for 6309 */
 	char		cpuclass;			/* processor class (CPU_* above) */
-	int			(*func)();			/* function */
+	int			(*func)(assembler *as, int opcode);			/* function */
 	rma_sect	permissible;		/* which RMA sections this opcode is allowed in */
 };
 
@@ -189,7 +192,7 @@ struct pseudo_opcode
 	char			*pseudo;        /* its name */
 	pseudo_class	class;
 	pseudo_info		info;
-	int				(*func)();		/* function */
+	int				(*func)(assembler *as);		/* function */
 	rma_sect		permissible;	/* which RMA sections this opcode is allowed in */
 };
 
